lrdd/row.go

```diff
@@ -0,0 +1,52 @@
+package lrdd
+
+import "fmt"
+
+type Row struct {
+	Key   string
+	Value MarshalUnmarshaler
+}
+
+type MarshalUnmarshaler interface {
+	MarshalMsg([]byte) ([]byte, error)
+	UnmarshalMsg([]byte) ([]byte, error)
+	ID() RowID
+	String() string
+}
+
+type RowID int32
+
+const (
+	RowIDBytes  RowID = 1
+	RowIDUint64 RowID = 2
+)
+
+func GetValue(rowID RowID) MarshalUnmarshaler {
+	switch rowID {
+	case RowIDBytes:
+		return &Bytes{}
+	case RowIDUint64:
+		i := Uint64(0)
+		return &i
+	default:
+		return rowTypes[rowID]()
+	}
+}
+
+func RegisterValue(rowID RowID, newFunc func() MarshalUnmarshaler) {
+	switch rowID {
+	case RowIDBytes:
+	case RowIDUint64:
+		panic("cannot register builtin row type")
```

Comment:
I think it would be clear to register builtin types in `rowTypes`, and unify the checking logic.

Reply:
To provide more kind error message, registering builtin types separately from the user-registered types and checking both maps is also fine.

Reply:
Done. 6fad2a5ed0d37b2e1d2f4565d44a846a5066eecc